### PR TITLE
Updated which command.

### PR DESCRIPTION
**Pull request by fcrespo82**
# Summary

Updated which command. Now it searches for the binary file inside all bin-paths of all plugins in the current version

Fixes: #205 

## Other Information

If it needs any improvement, please let me know!

### CODE REVIEW

File: lib/commands/which.sh [hunk 1]
```diff
@@ -1,51 +1,36 @@
...
+
+  if ls "$plugins_path" &> /dev/null; then
+    for plugin_path in "$plugins_path"/* ; do
+      plugin_name=$(basename "$plugin_path")
+      full_version=$(get_preset_version_for "$plugin_name")
+      # shellcheck disable=SC2162
+      IFS=' ' read -a versions <<< "$full_version"
+      for version in "${versions[@]}"; do
+        if [ -f "${plugin_path}/bin/exec-path" ]; then
+          echo "EXEC_PATH"
```
Comment (danhper) [inline lib/commands/which.sh]: I suppose this was for debugging purposes?

File: lib/commands/which.sh [hunk 2]
```diff
@@ -1,51 +1,36 @@
...
-    echo ""
-    echo "  asdf plugin-update $plugin_name"
-    echo ""
+  local command=$1
+  local plugins_path
+  plugins_path=$(get_plugin_path)
+
+  if ls "$plugins_path" &> /dev/null; then
+    for plugin_path in "$plugins_path"/* ; do
+      plugin_name=$(basename "$plugin_path")
```
Comment (danhper) [inline lib/commands/which.sh]: Please declare all these variables as `local`

File: lib/commands/which.sh [hunk 3]
```diff
@@ -1,51 +1,36 @@
...
+          cmd=$(basename "$executable_path")
+          install_path=$(find_install_path "$plugin_name" "$version")
+          executable_path="$("${plugin_path}/bin/exec-path" "$install_path" "$cmd" "$executable_path")"
+        fi
+        full_executable_path=$(get_executable_path "$plugin_name" "$version" "$executable_path")
+        local location
+        location=$(find "$full_executable_path" -name "$command" -type f -perm -u+x | sed -e 's|//|/|g')
+        if [ ! -z "$location" ]; then
+          echo "$location"
+          not_found=0
```
Comment (danhper) [inline lib/commands/which.sh]: I suppose we could break at this point? We only want the path of the executable actually used, not all the executable installed with this name.

File: lib/commands/which.sh [hunk 4]
```diff
@@ -1,51 +1,36 @@
...
+        if [ -f "${plugin_path}/bin/exec-path" ]; then
+          echo "EXEC_PATH"
+          cmd=$(basename "$executable_path")
+          install_path=$(find_install_path "$plugin_name" "$version")
+          executable_path="$("${plugin_path}/bin/exec-path" "$install_path" "$cmd" "$executable_path")"
+        fi
+        full_executable_path=$(get_executable_path "$plugin_name" "$version" "$executable_path")
+        local location
+        location=$(find "$full_executable_path" -name "$command" -type f -perm -u+x | sed -e 's|//|/|g')
+        if [ ! -z "$location" ]; then
```
Comment (danhper) [inline lib/commands/which.sh]: `! -z "$location"` -> `-n "$location"`

**Comment by fcrespo82** [comment]
I’ll work on this as soon as I can. Maybe today or tomorrow!! Thanks for the code review!

**Comment by fcrespo82** [comment]
If the version of a tool is set to system, what should we do? Just skip this plugin?

**Comment by Stratus3D** [comment]
It looks like tests are failing due to these changes.

**Comment by fcrespo82** [comment]
Yep, it failed on Linux. I have to get a Linux box to test it! I only have a macOS right now..

**Comment by fcrespo82** [comment]
Just fixed the error. Please check back to see if it is good to merge!

**Comment by fcrespo82** [comment]
Hy @danhper, any other change you want me to do? Or now it is good to merge?

**Comment by danhper** [comment]
@fcrespo82 Sorry for the delay.
Looks good, thank you very much!

**Comment by fcrespo82** [comment]
No problem at all! I know sometimes we get caught in the middle of something and leads to a rabbit hole making it difficult to find the time to do things that we care..

**Pull request merged**